test/parallel/test-stream-pipeline.js

```diff
@@ -1301,3 +1301,34 @@ const net = require('net');
     assert.strictEqual(res, '123');
   }));
 }
+
+{
+  const r = new Readable();
+  const t = new PassThrough();
+  const w = new Writable();
+
+  w.destroy();
+
+  pipeline([r, t, w], common.mustCall((err) => {
+    assert.strictEqual(err.code, 'ERR_STREAM_DESTROYED');
+    assert.strictEqual(err.message,
+                       'Cannot call write after a stream was destroyed');
```

Comment:
Can you make the error message more helpful? E.g.: Cannot call write after a 2 stream in pipeline() was destroyed.

Reply:
hmm, it doesn't make sense since `ERR_STREAM_DESTROYED` will cause 2 stream to be destroyed because destination stream is passed in destroyed state -- opposite logic here. Also I am not sure if `pipeline` really need to be mentioned in error message.

Reply:
First destroyed stream in passed arguments should cause all prepending streams to be destroyed

Reply:
Why only previous streams? All streams should be destroyed like what will happen when some stream is destroyed in the middle of pipeline processing.